aexpect/ops.py

```diff
@@ -473,3 +477,94 @@ def extract_tarball(session, tarball, target_dir, flags="-ap"):
+    :returns: statuses for each modified service
+    :rtype: [int]
+    """
+    statuses = []
+    for service in services:
+        status = session.cmd_status(f'service {service} status')
+        if status == 0:      # is running --> skip
+            LOG.debug(f'Service {service} already started - skipping')
+        elif status == 3:    # is stopped --> ok
+            LOG.debug(f'Service {service} stopped - starting')
+            status = session.cmd_status(f'service {service} start', timeout)
+        else:                # not sure, try restarting
+            LOG.debug(f'Querying status for service {service} resulted in '
+                      f'unexpected return code {status}. Try starting it')
+            status = session.cmd_status(f'service {service} start', timeout)
+        LOG.debug(f'Starting service {service} resulted in status {status}')
+        statuses += [status]
+    return statuses
+
+
+def stop_services(session, services, timeout=60):
+    """
+    Try stopping all services in given list.
+
+    :param session: virtual machine session
+    :type session: :py:class:`aexpect.ShellSession`
+    :param services: list of services to modify
+    :type services: [str]
+    :param int timeout: timeout for each service to stop
+    :returns: statuses for each modified service
+    :rtype: [int]
+    """
+    statuses = []
+    for service in services:
+        status = session.cmd_status(f'service {service} status')
+        if status == 0:      # is running --> stop
+            status = session.cmd_status(f'service {service} stop', timeout)
+        elif status == 3:    # is stopped --> ok
+            LOG.debug(f'Service {service} stopped already')
+        else:                # not sure, try stopping
+            LOG.debug(f'Querying status for service {service} resulted in '
+                      f'unexpected return code {status}. Try stopping it')
+            status = session.cmd_status(f'service {service} stop', timeout)
+        LOG.debug(f'Stopping service {service} resulted in status {status}')
+        statuses += [status]
+    return statuses
+
+
+def restart_services(session, services, timeout=60):
+    """
+    Try restarting all services in given list.
+
+    :param session: virtual machine session
+    :type session: :py:class:`aexpect.ShellSession`
+    :param services: list of services to modify
+    :type services: [str]
+    :param int timeout: timeout for each service to restart
+    :returns: statuses for each modified service
+    :rtype: [int]
+    """
+    statuses = []
+    for service in services:
+        status = session.cmd_status(f'service {service} status')
+        if status == 0:      # is running --> restart
+            status = session.cmd_status(f'service {service} restart', timeout)
+        elif status == 3:    # is stopped --> ok
+            LOG.debug(f'Service {service} stopped - starting')
+            status = session.cmd_status(f'service {service} start', timeout)
```

Comment:
Is it necessary to distinguish between those? The `service` tool ensures that already.